### PR TITLE
Support Go module

### PR DESCRIPTION
**Pull request by zchee**
Support Go Module.

[As I said before](https://github.com/neovim/go-client/pull/49#issuecomment-491496468), needs adding `go.mod`. Just added and fix travis test config.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/neovim/go-client
+
+go 1.12
```
Comment (justinmk) [inline go.mod]: I don't see this in https://github.com/golang/go/wiki/Modules#example . Does this mean we require 1.12? Should it be 1.11 instead?

Do we need a `require()` section?

Comment (zchee) [reply go.mod]: just remove it.

> Do we need a require() section?

The `require()` section needs if this repo depends on another third-party package. 
neovim/go-client written in only Go standard package. In this case, we don't need a `require`  section.

**Comment by zchee** [comment]
@justinmk commented and rebased. PTAL.

**Comment by zchee** [comment]
@justinmk I'll bump `v1.0.0`.

**Pull request merged**